lib/ui/popup.js

```diff
@@ -20,12 +21,27 @@ function Popup(opts) {
 
   this.beforePopupHide = this.beforePopupHide.bind(this);
   this.onResultsMouseOver = this.onResultsMouseOver.bind(this);
+  this.onFirstPopupOpen = this.onFirstPopupOpen.bind(this);
 }
 
 Popup.prototype = {
   init: function() {
     this.el = this.win.document.getElementById('PopupAutoCompleteRichResult');
     this.el.addEventListener('popuphiding', this.beforePopupHide);
+
+    // The first time the popup opens, a bit of popup state is not set
+    // correctly, causing no results to be shown, and causing the popup to fail
+    // to appear more than once (#138). To work around this bug, listen for the
+    // popup to be opened for the first time, and set the missing state.
+    //
```

Comment:
Yeah, that's a good point. I'll try to figure something out. Unfortunately, the source of this bug is that the `richlistitem` definition in `autocomplete.xml` looks like it's been copied into `urlbarBindings.xml` and slightly changed. Because the copying over and the additional changes are mixed in the same commit (https://github.com/mozilla/gecko-dev/commit/ee27759c), you have to wade through ~1800 lines of changes to try to make sense of what's been altered. I've been staring at this code for nearly a year now, on and off, and I wasn't able to make much progress--hence falling back to this PR that merely fixes the surface symptoms :-\

I'll probably link to the gecko-dev commit and add a warning comment that it is a large, complex diff.